schemas/+aod/trackMotion3D.m

```diff
@@ -0,0 +1,86 @@
+
+function [x,y,z,t] = trackMotion3D(fn, analysischan) 
+
+%% test function computeshifts
+% requires z:\libraries\matlab in path
+run(getLocalPath('/lab/libraries/hdf5matlab/setPath.m')) ;
+
+downsampleFactor = 1 ; % skip frames if > 1
+numFramesToAverageForRef = 10 ; % frames to average to determine the reference frame against which motion is computed
+useMultiCore = true ;
+postRefStartFrameTime = 0 ; % N secs to skip before analyzing motion signals, in many cases, right after starting the point scan, there is a slight drift
+
+br = aodReader(fn, 'Motion') ;
+br1 = AodMotionCorrectionReader(fn) ; % if motion was corrected by using aod offsets, this is where they would be read
+
+dat = br(:,:,:) ;
+numSamples = br1.sz ;
+duration = numSamples(:,1)/br1.Fs ; % sec
+
+motion_fs = size(dat,1)/duration ; % this is based on the temporal sampling rate
+delta_t = downsampleFactor/motion_fs ;
+
+numPlanes = br.planes ;
+coordinates = br.motionCoordinates ;
+
+% get pmt chan data
+dat2 = squeeze(dat(:,:,analysischan)) ;
+clearvars dat ;
+gridSize = sqrt(size(dat2,2)/numPlanes) ;
+
+% frame number to start from
+postRefStartFrame = max(round(postRefStartFrameTime*motion_fs),1) ;
+
+% chop off the first N secs
+dat2 = dat2(postRefStartFrame:end, :) ;
+
+needReference = true ; % first get a reference frame
+refparams = [] ;
+calibparams = [0 1; 0 1; 0 1] ; % dont apply any calibration
+shifts = zeros(size(dat2,1),3); % store motion vectors here
+dat2(1,:) = mean(dat2(1:numFramesToAverageForRef,:)) ; % average frames for computing reference
+[shifts(1,:), refparams] = computeshifts(dat2(1,:)',coordinates,gridSize,calibparams,refparams,needReference) ; % this position of the fitted sphere is treated as the reference
```

Comment:
I don't really like duplicating code in two places, but at the same time in this case I think it would be better than spanning dependencies between two pretty separate projects. It also avoids unintended side effects if one of the formats changes the other would break.